ads/google/imaVideo.js

```diff
@@ -221,241 +195,154 @@ let consentState;
 // Throttle for the showControls() function
 let showControlsThrottled = throttle(window, showControls, 1000);
 
+/**
+ * @param {!Node} parent
+ * @param {string} css
+ */
+function insertCss(parent, css) {
+  const style = parent.ownerDocument.createElement('style');
+  style./*OK*/ textContent = css;
+  parent.appendChild(style);
+}
+
+/**
+ * @param {string} state
+ * @param {boolean} active
+ */
+function toggleRootDataAttribute(state, active) {
+  const {'root': root} = elements;
+  const attributeName = `data-${state}`;
+  if (active) {
+    root.setAttribute(attributeName, '');
+  } else {
+    root.removeAttribute(attributeName);
+  }
+}
+
+/**
+ * @param {Document} elementOrDoc
+ * @return {!Object<string, !Element>}
+ */
+function renderElements(elementOrDoc) {
+  const html = htmlFor(elementOrDoc);
+
+  // Elements annotated with `ref="name"` are referenced as `elements['name']`.
+  // They also have their `ref` copied as a classname, so they can be selected
+  // from CSS using that exact value.
+  const root = html`
```

Comment:
We now have a single template that creates a full skeleton.

It minifies nicely on a distribution binary, likely smaller than the equivalent sequence of DOM calls:

```html
<div class=fill><div class=fill><video controlslist="nodownload nofullscreen noremoteplayback"playsinline ref=video></video>
</div><div ref=adContainer hidden class=fill></div><div ref=controls hidden><button ref=playButton></button><div class=countdownWrapper>
<div ref=countdown></div></div><div ref=time>-:- / 0:00</div><div ref=progress><div ref=progressLine></div><div ref=progressMarker></div>
</div><button ref=muteButton></button> <button ref=fullscreenButton></button></div><button ref=overlayButton class=fill></button></div>
```